rewrite-java/src/main/java/org/openrewrite/java/cleanup/NoValueOfOnStringType.java

```diff
@@ -64,12 +61,12 @@ public Duration getEstimatedEffortPerOccurrence() {
     protected JavaVisitor<ExecutionContext> getVisitor() {
         return new JavaVisitor<ExecutionContext>() {
             @Override
-            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
-                J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, executionContext);
+            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
                 if (VALUE_OF.matches(mi)) {
                     Expression argument = mi.getArguments().get(0);
-                    if ((argument instanceof J.Literal && TypeUtils.isOfType(argument.getType(), JavaType.Primitive.String))
```

Comment:
I don't think this is quite right.
Binary concatenation is implied to be a string by the compiler:
```java
int val = 1;
String temp = "a" + val; // valid
```
In this case, the method invocation is expecting a string:
```java
int val = 1;
String temp = "a" + val; // valid
void methodA(String a) {}
void methodB() {
  methodA(val); // invalid
  methodA(temp); // valid
}
```